### ISSUE TITLE
Use golang-jwt/jwt instead of form3tech-oss/jwt-go

### ISSUE THREAD

**Issue by dims** [kubernetes/kubernetes#104007]
We used to have dgrijalva/jwt-go a while ago which was unmaintained. So folks switched to a fork in `form3tech-oss` org. However there is a community maintained fork at https://github.com/golang-jwt/jwt we should be using.

Links:
- https://github.com/Azure/go-autorest/issues/642
- https://cloud-native.slack.com/archives/C01GVR8SY4R/p1627562086149400

```
vendor/github.com/heketi/heketi/client/api/go-client/client.go:	jwt "github.com/form3tech-oss/jwt-go"
vendor/github.com/Azure/go-autorest/autorest/adal/token.go:	"github.com/form3tech-oss/jwt-go"
vendor/go.etcd.io/etcd/server/v3/auth/jwt.go:	jwt "github.com/form3tech-oss/jwt-go"
vendor/go.etcd.io/etcd/server/v3/auth/options.go:	jwt "github.com/form3tech-oss/jwt-go"
```

So what do we need to do? Persuade the folks maintaining the following repos to switch to the community fork.
- https://github.com/heketi/heketi/
- https://github.com/Azure/go-autorest/
- https://github.com/etcd-io/etcd/

By first filing issues to raise their awareness and following it up with PRs

/area code-organization

**Comment by RinkiyaKeDad** [comment]
Looks like @Haimantika is already on it 🙂 thanks!

/unassign

**Comment by phlogistonjohn** [comment]
> Persuade the folks maintaining the following repos to switch to the community fork.

As I mentioned in heketi/heketi#1871  - I don't quite see what's better about this jwt fork than the previous jwt fork and I don't have the ability to investigate it deeply. If this is the consensus in kubernetes-land that's good enough for me. I am just unsure that, from the this issue alone, this where the rest of the k8s & the wider golang community is definitely going yet.  Pluse, the recent churn is a tad bit unsettling but I assume it will quiet down after a bit too.

Long story short, I'm ok with making the change but I feel like I/we lack background.

**Comment by dims** [comment]
@natalisucks any updates?

**Comment by dims** [comment]
related to https://github.com/kubernetes/kubernetes/issues/106422

**Comment by natalisucks** [comment]
Major apologies on the delay here, folks. Heketi maintainers ([as mentioned here](https://github.com/heketi/heketi/issues/1871)) are ready to look at a PR for this change, and I'm soon to create issues in the other two repositories. Will link once created.

**Comment by enj** [comment]
I think the intent here to to remove any mention of the old repo from k/k's go.mod / go.sum - once that is true I think we can close this and #106422 out.  We need to bump the versions of these deps that are used by k/k to include those fixes, which should cause the automatic removal of the old repo from our dep tree.  @natalisucks is that something you can look into?

**Comment by slaskawi** [comment]
Let me try to have a look at this one

/assign

**Comment by slaskawi** [comment]
Here's the status as of today:
* I created a WIP PR for updating all dependencies and get rid of form3tech-oss/jwt-go. You may find it here: https://github.com/kubernetes/kubernetes/pull/109501
* Unfortunately, I had to upgrade Heketi, which caused problems with Go modules (see https://github.com/heketi/heketi/issues/1897). I proposed a fix here: https://github.com/heketi/heketi/pull/1898 and we need to wait till the Heketi Team merge it. Once it's done, we'll need another release.

So currently the progress on this issue is **blocked**.
